include/swift/Parse/Token.h

```diff
@@ -108,7 +108,39 @@ class Token {
   bool isBinaryOperator() const {
     return Kind == tok::oper_binary_spaced || Kind == tok::oper_binary_unspaced;
   }
-  
+
+  /// Checks whether the token is either a binary operator, or is a token that
+  /// acts like a binary operator (e.g infix '=', '?', '.').
+  bool isBinaryOperatorLike() const {
+    if (isBinaryOperator())
+      return true;
+
+    switch (Kind) {
+    case tok::period:
```

Comment:
Yeah, that's because in SwiftSyntax, period can both be infix and postfix. Maybe we ought to at least add it to `OperatorLike` there. Or we could just remove it here as we don't actually rely on `tok::period` being matched here. I don't feel too strongly either way, any preference? A period does kind of seem "operator like".

Reply:
Oh hmm seems we rely on `OperatorLike` not matching period so that it's treated as unexpected for the `is not allowed in operator names` diagnostic in SwiftSyntax. Maybe we should just remove it here then

Reply:
Removed `tok::period` from here for now